Tools/autotest/arducopter.py

```diff
@@ -739,6 +739,84 @@ def fly_gcs_failsafe(self, side=60, timeout=360):
             self.reboot_sitl()
             raise ex
 
+    def fly_loiter_turns_auto(self, timeout=360):
+        # Test loiter turns in auto mode
+        ex = None
+        try:
+            num_wp = self.load_mission("copter_loiterturns_mission.txt")
+            self.change_mode('LOITER')
+            self.wait_ready_to_arm()
+            self.arm_vehicle()
+            self.change_mode('AUTO')
+            self.set_rc(3, 1550)
+            self.wait_waypoint(0, num_wp-1, timeout=500)
+            self.wait_disarmed()
+        except Exception as e:
+            self.print_exception_caught(e)
+            ex = e
+        self.set_rc_default()
+        self.reboot_sitl()
+        if ex is not None:
+            raise ex
+
+    def fly_loiter_turns_guided(self, timeout=360):
+        # Test loiter turns in guided
+        ex = None
+        try:
+            self.change_mode('LOITER')
+            self.wait_ready_to_arm()
+            self.arm_vehicle()
+            self.set_rc(3, 1800)
+            self.delay_sim_time(5)  # let it climb for few seconds
+            self.change_mode('GUIDED')
+            locations = [mavutil.location(-35.36249980, 149.16512400, 25, 0),
+                         mavutil.location(-35.36255880, 149.16393850, 30, 0),
+                         mavutil.location(-35.36249980, 149.16512400, 15, 0)]
+            radius = [10, 5, 20]
```

Comment:
It may rotate in place. You should also check the operation in a 0 meter radius.

ex:
https://youtu.be/dEeq11sonIQ?t=102

Reply:
Thanks @muramura for bringing this to my notice. I just found that we had a bug with the operation with 0 radius as described in issue #19028. I have fixed that bug and tested in too. It worked fine